Add `emptyfs` as image type

zentax-dev opened the pull request:
Depends on #1756 

This adds a new type to the image types. The `emptyfs` type makes it easy to add a slot with an empty filesystem. This can the be filled in a different way, for example with a hook.

jluebbe reviewed (commented):
This looks good, but should include some documentation and a test-case.

zentax-dev commented:
Rebased onto current master, added tests and documentation

zentax-dev commented:
Rebased to match the current state of master and #1756.
Moved the emmc test cases to its own file and refactored some of the source code to make it easier to read and maintain.

ejoerns commented on docs/reference.rst:
```diff
@@ -923,11 +923,34 @@ The following fields are supported for image sections:
+
   .. important::
-    RAUC uses the file name extension and the slot type to decide how to
+    If RAUC uses the file name extension and the slot type to decide how to
```
What is the 'If' for here?

jluebbe replied:
I've created https://github.com/rauc/rauc/pull/1812 for this.

ejoerns commented on docs/reference.rst:
```diff
@@ -1027,36 +1050,72 @@ In future releases, they will be accessible in other hooks, as well.
+  * ``ubifs``: UBIFS file system image
+  * ``squashfs``: SquashFS image (uncompressed, or compressed with one of ``lz4``, ``lzo``, ``xz``, ``xst``)
+  * ``emptyfs``: Creates an empty filesystem.
+  This type does not support setting a `filename` as no source data is needed.
```
The indentation looks wrong here if the sentence belongs to `emptyfs` image type.

zentax-dev commented on src/manifest.c:
```diff
@@ -108,19 +133,21 @@ static gboolean parse_image(GKeyFile *key_file, const gchar *group, RaucImage **
 
-	/* Setting the 'type' option for artifacts is not supported */
 	if (!iimage->artifact) {
+		/* Setting the 'type' option for artifacts is not supported.
```
created a new commit for this: #1814
